evalml/pipelines/component_graph.py

```diff
@@ -54,7 +54,10 @@ def from_list(cls, component_list, random_seed=0):
 
             component_dict[component_name] = [component_class]
             if previous_component is not None:
-                component_dict[component_name].append(f"{previous_component}.x")
+                if "sampler" in previous_component:
+                    component_dict[component_name].extend([f"{previous_component}.x", f"{previous_component}.y"])
```

Comment:
Need to track both X and y outputs from the undersampler (using 'sampler' to prepare for potential future oversampler transformers)

Reply:
Sounds good!